Standard-compliant parsing of Content-Type header (Fix #73)

Ecco opened the pull request.

bgoncal commented:
Hey @Ecco, thanks for the extensive debug around this issue, I had a quick glance over the situation and your PR looks fine, unfortunately I am leaving on vacations (for my wedding 🥳) today, and I won't have time to properly test and make a new build of the App before September 8th, I hope you understand.

Maybe @zacwest or @robbiet480 are around in the next days to take a look, but no promises.

robbiet480 commented:
This makes sense to me

bgoncal commented:
@Ecco can you please check the failing tests and lint? I'll manually test it throughout this week

bgoncal commented:
Your switch expect a string, cases will only accept what it expects, it could even be a function that returns a string.

Ecco commented:
- Added a commit to fix the linter check (for some reason check is broken on `main` branch)
- Moved away from Regex (apparently only available on macOS 13 and newer version of iOS).
- Ran linter, formatter, and tests locally.

Should be ok now, hopefully!

Ecco commented:
My bad, there was a typo in the last test (forgot to update the test identifier).
